contracts/liquidity_hub/pool-network/terraswap_router/src/contract.rs

```diff
@@ -215,24 +218,83 @@ fn assert_minimum_receive(
     Ok(Response::default())
 }
 
+fn add_swap_routes(
+    deps: DepsMut,
+    env: Env,
+    sender: Addr,
+    swap_routes: Vec<SwapRoute>,
+) -> Result<Response, ContractError> {
+    let contract_info = deps
+        .querier
+        .query_wasm_contract_info(env.contract.address)?;
+    if let Some(admin) = contract_info.admin {
+        if sender != deps.api.addr_validate(admin.as_str())? {
+            return Err(ContractError::Unauthorized {});
+        }
+    }
+
+    let mut attributes = vec![];
+
+    for swap_route in swap_routes {
```

Comment:
Very nice, was wondering why was this not in the fee_collector as I was going down. makes sense now